### PR TITLE
Fix success on invalid build

### LINKED ISSUE #9790: Build command exits with success with invalid arguments
### Describe the bug

Running `kc.sh build --cache nosuch` or any other invalid arguments exists with success (0) rather than failure (1).

### Version

main

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

```
$ bin/kc.sh build --cache nosuch && echo SUCCESS || echo FAILURE
Invalid value for option '--cache': nosuch. Expected values are: local, ispn
...
Use 'kc.sh build --help-all' to list all available options, including the start
options.
SUCCESS
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by DGuhr**
Closes #9790

### CODE REVIEW

**Comment by DGuhr** [comment]
New behaviour after this PR is in here: https://www.youtube.com/watch?v=c4bRyPbuvoY

@pedroigor @stianst Not beautiful, but works. it seems there is a problem with Quarkus.asyncExit - this _is_ called, but actually returns SUCCESS even when it has the right exit code as param, which it has now. Not sure if I am missing sth on our side, though, so we should doublecheck. When my assumption is right, this is indeed a quarkus bug and we should raise an issue for it, so we can remove the unwanted call to System.exit()

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java [hunk 1]
```diff
@@ -170,11 +192,13 @@ public String apply(String arg) {
             }
         });
 
-        cmd.execute(configArgsList.toArray(new String[0]));
+        exitCode = cmd.execute(configArgsList.toArray(new String[0]));
 
         if(!isDevMode()) {
             cmd.getOut().printf("Next time you run the server, just run:%n%n\t%s %s %s%n%n", Environment.getCommand(), Start.NAME, String.join(" ", getSanitizedRuntimeCliOptions()));
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java:198]: Does it make sense to only show this message when the exit code is success?

Comment (DGuhr) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java:187]: done

**Comment by stianst** [comment]
Should we also perhaps replace https://github.com/keycloak/keycloak/blob/main/quarkus/dist/src/main/content/bin/kc.sh#L1 with `#!/bin/bash -e` as then it would also result in exit code > 0 if something goes wrong in the script itself?

**Review by pedroigor — APPROVED** [review]

**Pull request merged**